client/src/components/CardDetail.js

```diff
@@ -0,0 +1,493 @@
+        open={open}
+        maxWidth={false}
+        classes={{ paper: classes.root }}
+        onSubmit={handleSubmit}
+      >
+        <DialogTitle
+          id="customized-dialog-title"
+          onClose={handleClose}
+          colorCode={colorCode}
+          setColorCode={setColorCode}
+        >
+          {card.title}
+        </DialogTitle>
+        <DialogContent dividers>
+          <Box component={"form"}>
+            <Grid container spacing={8}>
+              <Grid item className={classes.leftGrid}>
+                <Box component={'div'} mb={2}>
+                  <Typography className={classes.label}>Description</Typography>
+                  <TextField
+                    multiline
+                    rows={5}
+                    onChange={event => setDescription(event.target.value)}
+                    fullWidth
+                    variant={"outlined"}
+                    defaultValue={card.description}
+                    // onChange={(e) => e.target.description}
+                  />
+                </Box>
+                <Box component={'div'} display={showTag || "none"}  mb={2} >
+                  <Typography className={classes.label}>Tags</Typography>
+                  <TextField
+                    name={"tags"}
+                    placeholder={"Add tag ..."}
+                    size={"small"}
+                    defaultValue={tags}
+                    variant={"outlined"}
+                    onChange={event => { event.preventDefault(); setTags(event.target.value)}}
+                  />
+                </Box>
+                <Box component={'div'} mb={3} display={showChecklist || "none"}>
+                  <Typography className={classes.label}>Check-list</Typography>
+                  <ChecklistItems
+                    checklistItems={checklistItems}
+                    setChecklistItems={setChecklistItems}
+                  />
+                </Box>
+                <Box component={'div'} pb={3} display={showDeadline || "none"}>
+                  <Typography className={classes.label}>Deadline</Typography>
+                  <MuiPickersUtilsProvider utils={DateFnsUtils}>
+                    <DateTimePicker
+                      value={deadline}
+                      ampm={false}
+                      onChange={(date) => setDeadline(date)}
+                      animateYearScrolling
+                      clearable
+                    />
+                  </MuiPickersUtilsProvider>
+                </Box>
+                <Box component={'form'} mb={3} display={showAttachment || "none"} onSubmit={handleFileUpload}>
+                  <Typography className={classes.label}>Attachment</Typography>
+                  <Box component={'form'} variant={"outlined"}>
+                    <InputBase type={'file'} name={'file'}/>
+                    <Button type={'submit'} variant={'contained'} size={'small'} color={"secondary"}>Upload</Button>
```

Comment:
if the color for the button has been set as secondary, the text color need to be explicitly set to white